### ISSUE TITLE
Slack Ephemeral messages do not work.

### ISSUE THREAD

**Issue by dulfe** [microsoft/botbuilder-dotnet#5602]
## Version
4.13.2 (Latest in NuGet as of 2021-05-24)

## Describe the bug
Setting EPHEMERAL in a ChannelData for Slack messages does not work (it does not get sent)

## To Reproduce

```c#
      var activity = turnContext.Activity;
      var sm = new Microsoft.Bot.Builder.Adapters.Slack.Model.NewSlackMessage
      {
        Ephemeral = "YES",  // Any non-empty value
        Text = "text",
        Ts = activity.Timestamp?.DateTime.ToString(CultureInfo.InvariantCulture),
        Channel = activity.Conversation.Id,
        User = activity.Recipient.Id
      };

      var reply = new Activity();
      reply.ChannelData = sm;
      reply.Type = ActivityTypes.Message;

      await turnContext.SendActivityAsync(reply, cancellationToken);
```

## Expected behavior
An ephemeral message should be send to SLACK (visible only to the user)

## Screenshots
Expected:

![image](https://user-images.githubusercontent.com/12144048/119399656-24545900-bca7-11eb-9feb-73891dc57959.png)

## Additional context
USER is required as indicated in the documentation:

https://api.slack.com/methods/chat.postEphemeral

Current code does not send USER (file ```SlackClientWrapper.cs```):

```c#
            var data = new NameValueCollection
            {
                ["token"] = Options.SlackBotToken,
                ["channel"] = message.Channel,
                ["text"] = message.Text,
                ["thread_ts"] = message.ThreadTs,
            };
```

**Comment by stevkan** [comment]
@dulfe, thank you for your patience while I looked into this. It does appear that there is some sort of issue. It took a bit to get the correct configuration, but with the help of the '[Tester](https://api.slack.com/methods/chat.postEphemeral/test)' on their documentation page I was able to successfully send an ephemeral message to the Slack channel I and my bot were in. Using the same properties in the sample, I was not successful.

A couple notes about some of what I found. One, their [docs](https://api.slack.com/methods/chat.postEphemeral) state in the Arguments section and in the Tester that attachments are required, yet in the supporting text below Arguments it states it is optional. I tested with attachments. Two, I also only tested this in a channel (vs. an IM) as I can't think of a need for an ephemeral message where the conversation 1:1 with the bot.

I need to connect with a colleague regarding this which will likely happen after the holiday. I'll post an update as soon as I know more.

**Comment by dulfe** [comment]
@stevkan, before I even looked a Bot Builder, I was researching Slack's "Bolt for Javascript" and I "talked" with Slack Developer Support about ephemeral messages, they gave me a "sample" code for it which only has 3 fields in it.

**Sample:**
```javascript
await client.chat.postEphemeral({
          channel: payload.channel,
          user: slackUserId,
          text: `Sorry <@${slackUserId}>, you aren't registered in Acme. Please post in <#${helpChannelId}> for assistance.`
});
```

So I am guessing their "docs" are a little out-of-date :)

Thank you for looking into it.

**Comment by stevkan** [comment]
@dulfe, thank you for your patience. I'm still looking into this and hope to have an update either today or tomorrow for you.

**Comment by stevkan** [comment]
@dulfe, it appears the issue is in the `PostMessageAsync()` method in the adapter's SlackClientWrapper.cs file. In short, there is a [`data`](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs#L168) object that is assigned a number of properties that are essential to posting either a standard or ephemeral message. However, it is missing the `user` property and should look like this:

```dotnet
var data = new NameValueCollection
{
    ["token"] = Options.SlackBotToken,
    ["channel"] = message.Channel,
    ["text"] = message.Text,
    ["thread_ts"] = message.ThreadTs,
    ["user"] = message.User,
};
```

When testing, with the added property, I am able to successfully receive an ephemeral message from the bot.

![image](https://user-images.githubusercontent.com/18403946/121612475-55ef5300-ca0f-11eb-9274-a5c44fff79ff.png)

You can expect a fix to be applied to the adapter to resolve this in a future release. In the meantime, if you need the functionality, you can clone the dotnet repo, apply the same one line of code, build the projects, and then add the specific projects (Slack adapter plus dependencies) to your bot. This can serve as a workaround until the fix is applied and published. The steps I just mentioned are described in this [blog post](https://blog.botframework.com/2017/07/03/debug-bot-builder-source/). The post references our v3 SDK, but the steps are the same regardless.

Hope of help!

**Comment by dulfe** [comment]
@stevkan That is exactly what I am doing right now!..  Thank you for looking into it!

**Comment by dulfe** [comment]
@stevkan I tested your commit and it works with the sample given in the initial report (thank you!), but this issue was merged with #5604 which still does not work with your code.

In other words, setting  `Ephemeral` in `NewSlackMessage` works, but setting `reply.DeliveryMode = DeliveryModes.Ephemeral`  in an `Activity` does not.

I might be wrong in the assumption that the `Activity`'s `DeliveryMode` `Ephemeral` should work, if so, Could you tell me what is that for?

Thank you.
